doc/response-tagging.rst

```diff
@@ -19,19 +19,30 @@ The response tagger uses an instance of ``TagHeaderFormatter`` to know the
 header name used to mark tags on the content and to format the tags into the
 correct header value. This library ships with a
 ``CommaSeparatedTagHeaderFormatter`` that formats an array of tags into a
-comma-separated list. This format is expected for invalidation with the
-Varnish reverse proxy. When using the default settings, everything is created
-automatically and the ``X-Cache-Tags`` header will be used::
+comma-separated list. What kind of format you need obviously depends on what
+proxy you use and how this proxy is configured (if it even can be configured).
+It is important you do understand that the ``ResponseTagger`` is not directly
+linked to the proxy configuration so you are the one that has to make sure, they
+match. Don't worry too much though, the default settings will work out of the box
+and if you really need to adjust formatting, header names etc. you likely have
+some special environment that needs special adjustments. So for now, just remember
+you can configure all the stuff but you don't have to.
+
+For example, the :doc:`default configuration of Varnish <varnish-configuration>` in
+this documentation requires the header to contain a comma-separated list of tags
+and the default examples use the header ``X-Cache-Tags``. If you don't change
+the ``TagHeaderFormatter`` nor the header name, everything will be created
+automatically and work out of the box for you::
 
     use FOS\HttpCache\ResponseTagger;
 
     $responseTagger = new ResponseTagger();
 
 .. _response_tagger_optional_parameters:
 
-If you need a different behavior, you can provide your own implementation of
-the ``TagHeaderFormatter`` interface. But be aware that your
-:ref:`Varnish configuration <varnish_tagging>` has to match with the tag on the response.
+As explained already, if you need a different behavior, you can provide your own
+implementation of the ``TagHeaderFormatter`` interface. But again, be aware that your
```

Comment:
lets not add "again".

maybe "provide your own ``TagHeaderFormatter`` instance"? does not need to be a custom implementation of the interface, like in the example below. (i know, that is just reformatting, but while we touch it...)